pkg/ccl/backupccl/backup_job.go

```diff
@@ -714,7 +714,14 @@ func (b *backupResumer) Resume(ctx context.Context, execCtx interface{}) error {
 			return errors.Wrap(err, "failed to run backup")
 		}
 
-		log.Warningf(ctx, `BACKUP job encountered retryable error: %+v`, err)
+		// If we are draining, it is unlikely we can start a
+		// new DistSQL flow. Exit with a retryable error so
+		// that another node can pick up the job.
+		if p.ExecCfg().JobRegistry.IsDraining() {
```

Comment:
I started to wonder if we should just add this conditional to the registry state machine, to ignore just _any_ returned errors  from Resume if the registry is draining when they're returned rather than persist them and go to failing, but I guess that could potentially ignore a should-be-fatal error?

probably less backport-safe too, so good with this for now and then debate that later

Reply:
I almost went that way as well since it definitely feels like there is something more holistic to do here.  I was also worried about should-be-fatal errors. But, as I think about it some more, a should-be-fatal error will presumably happen again when the job is retried -- so perhaps that wouldn't be that much of a concern.  But, my expectation here is that there might be more problems with draining to tease out as these tests run for a bit.